middleware/headers.go

```diff
@@ -0,0 +1,65 @@
+package middleware
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/labstack/echo/v4"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+/*
+   Parse the required headers for processing this request. Currently this
+   involves _three_ major headers:
+
+   1. `x-rh-identity`: contains the account number and various other information
+      about the request. This is set by 3scale.
+
+   2. `x-rh-sources-psk`: a pre-shared-key (psk) which is used internally to
+      authenticate from within the CRC cluster. This is checked against a list
+      of known keys which are set in vault, if it matches any of them the
+      request is authorized.
+
+    3. `x-rh-sources-account-number`: used with a PSK to access a certain
+       account. Only accessible from within the CRC cluster.
+*/
+func ParseHeaders(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		// the PSK related headers - just storing them as raw strings.
+		if c.Request().Header.Get("x-rh-sources-psk") != "" {
+			c.Set("psk", c.Request().Header.Get("x-rh-sources-psk"))
+		}
+
+		if c.Request().Header.Get("x-rh-sources-account-number") != "" {
+			c.Set("psk-account", c.Request().Header.Get("x-rh-sources-account-number"))
+		}
+
+		// parsing the base64-encoded identity header if present
+		if c.Request().Header.Get("x-rh-identity") != "" {
+			// store it raw first.
+			c.Set("x-rh-identity", c.Request().Header.Get("x-rh-identity"))
```

Comment:
A nitpick: do we really need to store it raw? The only retrieval of the raw "x-rh-identity" header that I've found in the repository is in [middleware/authorization.go#L48](https://github.com/RedHatInsights/sources-api-go/blob/main/middleware/authorization.go#L48). I'd suggest not storing "x-rh-identity" since it's never used.